pkg/proxy/apis/config/validation/validation.go

```diff
@@ -317,3 +318,12 @@ func validateShowHiddenMetricsVersion(version string, fldPath *field.Path) field
 
 	return allErrs
 }
+
+func validateInterfacePrefix(interfacePrefix string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	const maxIfNameSize = 16 // linux/if.h (as defined by IFNAMSIZ in kernel)
```

Comment:
this will allow to match only one interface.
Also, if I set `--detect-local-mode` to localInterface and I forget to set the interfacePrefix it will match all the interfaces in the host.

@thockin @danwinship what do you think of make the validation something like:

```
if detect-mode == local-interface 
    validate interfaceName is > 3  < 14
```
minimum because `eth` == 3
maximum 14 because it leaves us with 2 free slots that is enough since names are alphanumerical

Reply:
ack yeah I am ok with `>= 3 && <14`.

Reply:
it is better if you wait for Tim's and Dan's opinion :)

Reply:
> this will allow to match only one interface. Also, if I set `--detect-local-mode` to localInterface and I forget to set the interfacePrefix it will match all the interfaces in the host.

actually that isn't true right? if I set `--detect-local-mode=Interface`, and interfacePrefix isn't set, it will default to no-op detect-local. A prefix has to be set if one has to use the interface mode for local traffic detector.

Also this brings me to the next question, should I be making the InterfacePrefix accept a string of comma separated names to allow match on multiple interface prefixes?

Reply:
I don't think there's much benefit in validating the length of the interface name. If the user gets this argument wrong, it's likely to be because of a typo (`--interface-prefix veht`) not because they passed a string longer than 16 characters...

However, you should validate that if `--interface-prefix` is non-empty if and only if `--detect-local-mode` is `Interface`.

And in that case, you shouldn't have to re-validate that it's non empty in the proxy code.

Reply:
ack, will change it per Dan's suggestion.

Reply:
qq, will we always go to `getLocalDetector` code in kube-proxy only after we pass the validation? Asking because for the rest of the parameters like podCIDR/nodeCIDR we seem to revalidate their length again in the proxy code to ensure they are not empty.